homeassistant/components/sensor/ambient_station.py

```diff
@@ -19,72 +18,66 @@
 
 REQUIREMENTS = ['ambient_api==1.5.2']
 
+_LOGGER = logging.getLogger(__name__)
+
 CONF_APP_KEY = 'app_key'
+CONF_UNITS = 'units'
 
 SENSOR_NAME = 0
 SENSOR_UNITS = 1
 
-CONF_UNITS = 'units'
 UNITS_US = 'us'
 UNITS_SI = 'si'
 UNIT_SYSTEM = {UNITS_US: 0, UNITS_SI: 1}
 
-SCAN_INTERVAL = timedelta(seconds=300)
+DEFAULT_APP_KEY = '32f561c4cb3a400d9c71ae0e96495466beaea220e315403c955b8f2bb' \
```

Comment:
We can't embed authentication keys in our source code. This means the key is public, and defeats the purpose of having a key at all.

Reply:
Ambient Weather's API requires an application key (designed to be shared by _all_ instances of an implementing application) and an API key (designed to be provided by each user). If users have to provide an API key, I don't see how this is insecure – nothing can be done with the application key alone. What am I missing?

Reply:
By exposing this key to the public, you've removed Ambient's ability to track what applications are creating API requests, since any app could make use of this key now. We don't embed secrets in our public source code.

Reply:
Additionally, please see [this issue](https://github.com/home-assistant/home-assistant/issues/19932) and note the first comment:

>Ambient weather does not want to provide app keys and has told me the developer of this should have hard coded the app key in to this.
>
>I referred them to the HA documentation and the following was the response:
>
>"Unfortunately, he is wrong. It is one app with many users, not many apps with many users. Thus, he needs to supply you with the app key unique to his application."